pylint/checkers/variables.py

```diff
@@ -1282,8 +1282,31 @@ def leave_functiondef(self, node: nodes.FunctionDef) -> None:
 
         global_names = _flattened_scope_names(node.nodes_of_class(nodes.Global))
         nonlocal_names = _flattened_scope_names(node.nodes_of_class(nodes.Nonlocal))
+        comprehension_target_names = []
+
+        def find_assigned_names_recursive(
```

Comment:
I always mess up my declaration order, but doesn't this get redefined for every `FunctionDef` we visit? That might become performance impacting?

Reply:
Yeah I can take a look at pulling this out into a private instance method.

Reply:
It was tested. But I'm adding another case with more nesting.

> I always mess up my declaration order, but doesn't this get redefined for every FunctionDef we visit? That might become performance impacting?

Now that I'm looking closely, I guess I don't see a performance impact. Do nested functions really have a worse impact over nested loops and such?

Reply:
I really don't know 😄 

I'm not sure if `find_assigned_names_recursive` is stored in memory or if it is re-declared every time `leave_functiondef` gets called. The later sounds as if that could be problematic.